### PR TITLE
Qstls bugfix relating to adr(x,0)

### PR DESCRIPTION
**Pull request by FotisKalk**

### CODE REVIEW

File: src/qstls.cpp [hunk 1]
```diff
@@ -335,6 +335,11 @@ int Qstls::checkAdrFixed(const vector<double> &wvg_,
   const bool consistentMatsubara = nl_ == in.getNMatsubara();
   const bool consistentTheta = abs(Theta_ - in.getDegeneracy()) <= tol;
   const bool consistentGrid = wvg_.size() == wvg.size() && wvgMaxDiff <= tol;
+  if (!consistentMatsubara) {
+    std::cout << "Inconsistent Matsubara" << std::endl;
```
Comment (fedluc) [inline src/qstls.cpp:339]: Use std::cerr here so the error message doesn't get buffered

File: src/qstls.cpp [hunk 2]
```diff
@@ -603,19 +608,20 @@ double AdrFixed::integrand1(const double &q, const double &l) const {
 double
 AdrFixed::integrand2(const double &t, const double &y, const double &l) const {
   const double q = itg.getX();
-  if (q == 0 || t == 0 || y == 0) { return 0; };
```
Comment (fedluc) [inline src/qstls.cpp:606]: Are you sure that we can safely remove this line?

File: src/vector3D.cpp [hunk 3]
```diff
@@ -61,6 +61,11 @@ void Vector3D::fill(const double &num) {
   std::for_each(v.begin(), v.end(), [&](double &vi) { vi = num; });
 }
 
+void Vector3D::fill(const size_t i, const double &num) {
+  const auto dest = v.begin() + i * s2 * s3;
```
Comment (fedluc) [inline src/vector3D.cpp:65]: Make dest a constant reference (as we do in line 70)

File: src/qstls.cpp (hunk 2 above)
Comment (FotisKalk) [reply src/qstls.cpp:606]: There seems to be no singularity for either q = 0 , y = 0 (w in the expression Panagiotis has) that's why I removed it. As for t = 0 singularities for both expressions adr(x,l), adr(x,0) appear only when x = y. I added the corresponding analytical limits in the two new if statements. I also tested it and didn't see any issues.

**Pull request merged**